### DISCUSSION TITLE
Error message when change the input file in phase_field-mechanics$

### DISCUSSION THREAD

**Discussion by zengfy-hust** [idaholab/moose#20842] [Q&A Modules: Phase field] [answered]
Hello MOOSE workers ：
        I recently find an module just suitable for my research ,phase_field-mechanics.which provide the input options of euler angles .howerer ,since there is a vacant in introduction documentation of combined modules .I wonder what input format  should I change to be suitable for grn_36_test_2D.tex .I change the file into grain file exported from OIM ,and the following errdor message occured > I attached the file I used in this module . 
*** ERROR ***
The following error occurred in the object "hex_ic", of type "PolycrystalHex".

PolycrystalHex requires a square or cubic number depending on the mesh dimension

Stack frames: 16
0: libMesh::print_trace(std::ostream&)
1: moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
2: callMooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, MooseApp*)
3: /home/zengfanyi/projects/moose/modules/phase_field/lib/libphase_field-opt.so.0(+0x28ae74) [0x7f00a2f90e74]
4: PolycrystalHex::precomputeGrainStructure()
5: PolycrystalUserObjectBase::execute()
6: FEProblemBase::joinAndFinalize(TheWarehouse::QueryCache<>, bool)
7: FEProblemBase::computeUserObjectsInternal(MooseEnumItem const&, Moose::AuxGroup const&, TheWarehouse::QueryCache<>&)
8: FEProblemBase::computeUserObjects(MooseEnumItem const&, Moose::AuxGroup const&)
9: FEProblemBase::initialSetup()
10: Transient::init()
11: MooseApp::executeExecutioner()
12: MooseApp::run()
13: /home/zengfanyi/projects/moose/modules/combined/combined-opt(+0x527e) [0x561245c7127e]
14: __libc_start_main
15: /home/zengfanyi/projects/moose/modules/combined/combined-opt(+0x550d) [0x561245c7150d]
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor
 can  you help me fix this .
[test.zip](https://github.com/idaholab/moose/files/6505750/test.zip)

**Comment by GiudGiud** [comment, marked as answer]
@zengfy-hust any update on this?
Issues arent tracked the same way so we did not see this. 
You re probably just missing a parameter here

**Reply by laagesen** [reply]
This IC tries to fit n x n hexagons in a 2D simulation, where n is an integer, or n x n x n in 3D. So, you need to supply a grain number that is a square of that integer for a 2D simulation or the cube of that integer for 3D. So, for a 2D simulation, you need to supply grain_num = 4, 9, 16 etc. to get 2 x 2, 3 x 3, 4 x 4 etc.
